apps/api/src/app/messages/usecases/get-messages/get-messages.usecase.ts

```diff
@@ -47,26 +47,15 @@ export class GetMessages {
       }
     }
 
-    const totalCount = await this.messageRepository.count(
-      query,
-      COUNT_LIMIT
-      /*
-       * todo NV-2161 in version 0.16:
-       *  update option as below,
-       *  update below:  hasMore = feed.length < totalCount
-       *  remove totalCount
-       * { skip: command.page * LIMIT, limit: LIMIT + 1 }
-       */
-    );
-
-    const hasMore = this.getHasMore(command.page, LIMIT, data.length, totalCount);
+    const totalCount = await this.messageRepository.count(query, COUNT_LIMIT);
 
     return {
       page: command.page,
-      totalCount,
-      hasMore,
+      hasMore: data.length < totalCount,
       pageSize: LIMIT,
       data,
+      skip: command.page * LIMIT,
+      limit: LIMIT + 1,
```

Comment:
Those two should be in the getCount query options. and not in the return value.

Regarding the totalCount removal, I could love to do quick sync :)